lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
1. If we need to do this, lets add the code disabled by default.
2. We can use JobObject to enforce memory limit on the process and see if that would make GC work better? I am not entirely sure what the ASA experience here was. I can provide you code to help you write this if you want to try this out.
3. Who is killing the process when the memory is going above 8GB?
4. Is it possible to reuse objects/generate less garbage?